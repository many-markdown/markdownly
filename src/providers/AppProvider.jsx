import React, { useReducer, useEffect } from 'react';
import { AppContext } from '../hooks/appContext';
import reducer, { initialState } from '../reducers/appReducer';
import { setInitialState } from '../actions/markdownActions';

// eslint-disable-next-line react/prop-types
export default function AppProvider({ children }) {

  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    const initialState = localStorage.getItem('FILE_STORAGE'); 
    dispatch(setInitialState(JSON.parse(initialState)));
  }, []);

  return (
    <AppContext.Provider value={{ state, dispatch }} >
      {children}
    </AppContext.Provider>
  );
}
