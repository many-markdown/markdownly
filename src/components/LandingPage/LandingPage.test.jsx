import React from 'react';
import { shallow } from 'enzyme';
import AppProvider from '../../providers/AppProvider';
import LandingPage from './LandingPage.jsx';

describe('Landing page component', () => {
  it('renders landing page', () => {
    const wrapper = shallow(<AppProvider><LandingPage /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
