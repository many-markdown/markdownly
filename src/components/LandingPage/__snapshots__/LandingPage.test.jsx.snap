// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component renders landing page 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "currentId": 1,
        "markdownFileList": Array [
          Object {
            "body": "# hello there",
            "fileName": "default",
            "id": 1,
          },
        ],
        "searchCategory": "fileName",
        "searchQuery": "",
      },
    }
  }
>
  <LandingPage />
</ContextProvider>
`;
